File: lib/star-data/repository/starDataRepository.ts
import type { StarDataItem } from "../models/starDataItem";
import type { StarDataPack } from "../models/starDataPack";
import type { StarDataSummary } from "../models/starDataSummary";

export interface StarDataRepository {
  fetchItems(params: {
    username: string;
    date: string;
    category?: string;
    genre?: string;
  }): Promise<StarDataItem[]>;

  fetchPack(packId: string): Promise<StarDataPack>;

  fetchSummary(username: string): Promise<StarDataSummary>;

  hideItem(itemId: string): Promise<void>;
}

File: lib/src/features/star_data/data/star_data_repository.dart
import '../models/star_data_item.dart';
import '../models/star_data_pack.dart';
import '../models/star_data_summary.dart';

abstract class StarDataRepository {
  Future<List<StarDataItem>> fetchItems({
    required String username,
    required DateTime date,
    String? category,
    String? genre,
  });

  Future<StarDataPack> fetchPack(String packId);

  Future<void> hideItem(String itemId);

  Future<StarDataSummary> fetchSummary(String username);
}
